ui/guest/src/components/GuestProxy.js

```diff
@@ -250,46 +282,63 @@ export function GuestProxy(props) {
+                switch (field.type) {
+                  case 'repeat':
+                  case 'node-selector': {
+                    throw new Error('Not implemented.');
+                  }
+                  default:
+                    $clone.find(`[data-craftercms-field-id="${id}"]`).html(instance[id]);
                 }
-                default:
-                  $clone.find(`[data-craftercms-field-id="${id}"]`).html(instance[id]);
-              }
-            });
-          };
-
-          processFields(instance, contentType.fields);
-
-          const $daddy = $(`[data-craftercms-model-id="${modelId}"][data-craftercms-field-id="${fieldId}"]:not([data-craftercms-index])`);
-
-          const $siblings = $daddy.find('> *');
-
-          if ($siblings.length === targetIndex) {
-            $daddy.append($clone);
+              });
+            };
+            processFields(instance, contentType.fields);
           } else {
-            $clone.insertBefore($siblings.eq(targetIndex));
+            $clone = $(`<div  data-craftercms-model-id="${modelId}" data-craftercms-field-id="${fieldId}">${instance.craftercms.label}</div>`);
           }
 
-          forEach(
-            $daddy.children(),
-            (el, i) => {
+          insertElement($clone, $daddy, targetIndex);
 
-              $(el).attr('data-craftercms-index', i);
+          updateElementRegistrations(Array.from($daddy.children()), 'insert', targetIndex);
 
-              const pr = ElementRegistry.fromElement(el);
-
-              pr && context.deregister(pr.id);
-              registerElement(el);
+          break;
+        }
+        case INSERT_INSTANCE_OPERATION: {
+          const { modelId, fieldId, targetIndex, instance } = op.args;
+
+          const $spinner = $(`
+            <svg class="craftercms-placeholder-spinner" width=50 height=50 viewBox="0 0 50 50" xmlns="http://www.w3.org/2000/svg">
+              <circle class="path" fill="none" stroke-width=5 stroke-linecap="round" cx="25" cy="25" r="20"/>
+            </svg>
+          `);
+
+          const $daddy = getDropzoneElement(modelId, fieldId, targetIndex);
+
+          insertElement($spinner, $daddy, targetIndex);
+
+          const id = Date.now();
+
+          message$.pipe(
+            filter((e) => (e.data?.type === COMPONENT_INSTANCE_HTML_RESPONSE) && (e.data?.payload.id === id)),
+            map(e => e.data),
+            take(1)
+          ).subscribe(function ({ payload }) {
+            const $root = $('<div/>').html(payload.response);
+            const $component = ($root.children().length > 1) ? $root : $($root.children()[0]);
+            $component.attr('data-craftercms-model-id', modelId);
+            $component.attr('data-craftercms-field-id', fieldId);
+            $spinner.remove();
+            insertElement($component, $daddy, targetIndex);
+            updateElementRegistrations(Array.from($daddy.children()), 'insert', targetIndex);
```

Comment:
Consider nested cases.

Index may be `0.1`. You'd need to send it the last piece (possibly as an int).

Check other usages of `updateElementRegistrations`.

Reply:
remember you can also use `$().children().toArray()`